node/src/blockchain/blockchain_interface/blockchain_interface_web3/lower_level_interface_web3.rs

```diff
@@ -1,395 +1,762 @@
+                    H256::from_str(
+                        "3dc91b98249fa9f2c5c37486a2427a3a7825be240c1c84961dfb3063d9c04d50"
+                    )
+                    .unwrap()
+                ),
+                transaction_index: Some(U64::from(29)),
+                log_index: Some(U256::from(29)),
+                transaction_log_index: None,
+                log_type: None,
+                removed: Some(false),
+            }
         );
-        let send_params = send_params_arc.lock().unwrap();
-        let rpc_call_params = vec![
-            Value::String(String::from("0x5c361ba8d82fcf0e5538b2a823e9d457a2296725")),
-            Value::String(String::from("pending")),
-        ];
-        let expected_request =
-            web3::helpers::build_request(1, "eth_getTransactionCount", rpc_call_params);
-        assert_eq!(*send_params, vec![(1, expected_request)])
     }
 
     #[test]
-    fn low_interface_web3_get_transaction_id_handles_err() {
-        let act = |subject: &LowBlockchainIntWeb3<Http>, wallet: &Wallet| {
-            subject.get_transaction_id(wallet)
-        };
-
-        assert_error_from_unintelligible_response(act, "invalid hex character")
-    }
-
-    fn assert_error_from_unintelligible_response<F>(act: F, expected_err_fragment: &str)
-    where
-        F: FnOnce(&LowBlockchainIntWeb3<Http>, &Wallet) -> ResultForBalance,
-    {
+    fn get_transaction_logs_fails() {
         let port = find_free_port();
-        let _test_server = TestServer::start (port, vec![
-            br#"{"jsonrpc":"2.0","id":0,"result":"0x000000000000000000000000000000000000000000000000000000000000FFFQ"}"#.to_vec()
-        ]);
-        let (_event_loop_handle, transport) = Http::with_max_parallel(
-            &format!("http://{}:{}", &Ipv4Addr::LOCALHOST.to_string(), port),
-            REQUESTS_IN_PARALLEL,
-        )
-        .unwrap();
-        let chain = TEST_DEFAULT_CHAIN;
-        let wallet = Wallet::from_str("0x3f69f9efd4f2592fd70be8c32ecd9dce71c472fc").unwrap();
-        let subject = make_subject(transport, chain);
+        let _blockchain_client_server = MBCSBuilder::new(port)
+            .raw_response(r#"{
+              "jsonrpc": "2.0",
+              "id": 1,
+              "result": [
+                {
+                  "address": "0x0000000000000000000000000070617965655f31",
+                  "blockHash": "0x7c5a35e9cb3e8ae0e221ab470abae9d446c3a5626ce6689fc777dcffcab52c70",
+                  "blockNumber": "0x5c29fb",
+                  "data": "0x0000000000000000000000003e331",
+                  "logIndex": "0x1d",
+                  "removed": false,
+                  "topics": [
+                    "0x241ea03ca20251805084d27d4440371c34a0b85ff108f6bb5611248f73818b80"
+                  ],
+                  "transactionHash": "0x3dc91b98249fa9f2c5c37486a2427a3a7825be240c1c84961dfb3063d9c04d50",
+                  "transactionIndex": "0x1d"
+                }
+              ]
```

Comment:
Please elevate somehow the chances the reader checking this test will discover why this data sent back will provoke this error. You are responsible for leaving the test in a state in which it wouldn't be too hard to figure it out, or rephrased, that he won't be in doubts about it and that's what just happened to me. I think it's specially unclear for this test, otherwise I'd hold back complaining.